tests/integration/dbapi/async/V2/test_streaming.py

```diff
@@ -0,0 +1,123 @@
+
+        await c.execute_stream(all_types_query)
+        assert len(await c.fetchmany(0)) == 0, "Invalid data size returned by fetchmany"
+        data = await c.fetchmany()
+        assert len(data) == 1, "Invalid data size returned by fetchmany"
+        assert_deep_eq(
+            data, all_types_query_response, "Invalid data returned by fetchmany"
+        )
+
+
+async def test_streaming_multiple_records(
+    connection: Connection,
+) -> None:
+    """Select handles multiple records properly."""
+    row_count, value = (
+        100000,
+        "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+    )
+    sql = f"select '{value}' from generate_series(1, {row_count})"
+
+    async with connection.cursor() as c:
+        await c.execute_stream(sql)
+        assert c.rowcount == -1, "Invalid rowcount value before fetching"
+        async for row in c:
+            assert len(row) == 1, "Invalid row length"
+            assert row[0] == value, "Invalid row value"
+        assert c.rowcount == row_count, "Invalid rowcount value after fetching"
+
+
+def get_process_memory_mb() -> float:
+    """Get the current process memory usage in MB."""
+    return psutil.Process(os.getpid()).memory_info().rss / (1024**2)
+
+
+@mark.slow
+async def test_streaming_limited_memory(
+    connection: Connection,
+) -> None:
+
+    memory_overhead_threshold_mb = 100
+    row_count, value = (
+        10000000,
+        "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+    )
+    original_memory_mb = get_process_memory_mb()
+    sql = f"select '{value}' from generate_series(1, {row_count})"
+    async with connection.cursor() as c:
+        await c.execute_stream(sql)
+
+        memory_diff = get_process_memory_mb() - original_memory_mb
+        assert (
+            memory_diff < memory_overhead_threshold_mb
+        ), f"Memory usage exceeded limit after execution (increased by {memory_diff}MB)"
+
+        assert c.rowcount == -1, "Invalid rowcount value before fetching"
+        async for row in c:
+            assert len(row) == 1, "Invalid row length"
+            assert row[0] == value, "Invalid row value"
+        assert c.rowcount == row_count, "Invalid rowcount value after fetching"
+
+        memory_diff = get_process_memory_mb() - original_memory_mb
+        assert (
+            memory_diff < memory_overhead_threshold_mb
+        ), f"Memory usage exceeded limit after fetching results (increased by {memory_diff}MB)"
+
+
+async def test_streaming_error(
+    connection: Connection,
+) -> None:
+    """Select handles errors properly."""
+    sql = (
+        "select date(a) from (select '2025-01-01' as a union all select 'invalid' as a)"
+    )
+    async with connection.cursor() as c:
+        with raises(FireboltStructuredError) as e:
+            await c.execute_stream(sql)
+
+        assert "Unable to cast TEXT 'invalid' to date" in str(
+            e.value
+        ), "Invalid error message"
```

Comment:
@bogdantruta-firebolt in his [PR](https://github.com/firebolt-db/firebolt-node-sdk/pull/137/files#diff-1f34f537b7094dc5b4529a89c412d73a87e6d23d7e4b7df2e744a22ad9a3f218) found a nice way to test a request that returns 200, but fails while being processed. Let's use that in another test here too to make sure we raise correct errors during streaming:
```
select 1/(i-100000) as a from generate_series(1,100000) as i
```